src/jvm/main/org/jetbrains/kotlinx/lincheck/trace/TraceCompressor.kt

```diff
@@ -189,6 +207,121 @@ private fun SingleThreadedTable<TraceNode>.compressThreadStart() = compressNodes
     newNode
 }
 
+/**
+ * Remove $iv from owner names, due to inline functions.
+ */
+private fun SingleThreadedTable<TraceNode>.compressInlineIV() = compressNodes { node ->
+    if (node !is CallNode || node.tracePoint.ownerName == null) return@compressNodes node
+    node.tracePoint.ownerName = node.tracePoint.ownerName!!.removeSuffix("\$iv")
+    node
+}
+
+/**
+ * Removes $this. owner names.
+ */
+private fun SingleThreadedTable<TraceNode>.compressDollarThis() = compressNodes { node ->
+    if (node !is CallNode || node.tracePoint.ownerName == null) return@compressNodes node
+    if (node.tracePoint.ownerName == "\$this") node.tracePoint.ownerName = null
+    node
+}
+
+/**
+ * Removes nested class dollars from owner names.
+ */
+private fun SingleThreadedTable<TraceNode>.replaceNestedClassDollar() = compressNodes { node ->
+    if (node is CallNode && node.tracePoint.ownerName != null) {
+        val newOwner = fixNestedClassDollar(node.tracePoint.ownerName!!)
+        node.tracePoint.ownerName = newOwner
+    }
+    node
+}
+
+/**
+ * Removes synthetic field due to lambda capture. $fieldName.element -> fieldName
+ */
+private fun SingleThreadedTable<TraceNode>.compressLambdaCaptureSyntheticField() = compressNodes { node ->
+    if (node is EventNode 
+        && node.tracePoint is ReadTracePoint
+        && node.tracePoint.ownerRepresentation?.startsWith("$") == true
+        && node.tracePoint.fieldName == "element"
+    ) {
+        node.tracePoint.fieldName = node.tracePoint.ownerRepresentation!!.removePrefix("$")
+        node.tracePoint.ownerRepresentation = null
+    }
+    
+    if (node is EventNode 
+        && node.tracePoint is WriteTracePoint
+        && node.tracePoint.ownerRepresentation?.startsWith("$") == true
+        && node.tracePoint.fieldName == "element"
+    ) {
+        node.tracePoint.fieldName = node.tracePoint.ownerRepresentation!!.removePrefix("$")
+        node.tracePoint.ownerRepresentation = null
+    }
+    node
+}
+
+private fun SingleThreadedTable<TraceNode>.compressVolatileDollar() = compressNodes { node ->
```

Comment:
Please add documentation explaining what this function does and an example of before/after trace change. Also, add a link to the corresponding issue: https://github.com/JetBrains/lincheck/issues/687